astropy/io/fits/hdu/table.py

```diff
@@ -909,11 +909,19 @@ def _writedata_by_row(self, fileobj):
         for idx in range(len(self.data)):
             for field in fields:
                 item = field[idx]
+                field_width = None
 
                 if field.dtype.kind == 'U':
+                    i = field.dtype.str.index(field.dtype.kind)
```

Comment:
That looks like a real hackish way to get the number from the string representation of the dtype. You could maybe add that as an inline comment so the next one knows what this is about.